all: add testshort target that skips slow tests

RaduBerinde opened the pull request:
Adding `make testshort` which adds the `-short` flag to `make test`. Disabling
tests over 10 seconds in this mode (except the main SQL logic test suite).

A reasonable workflow would be to run `make test PKG=./pkg/foo` for whatever
package you're touching and run `make testshort` for the rest.

Fixes #13025.

Also:
Add `make checkshort` which is `make check` with TestMetacheck and TestUnconvert
disabled (these tests are slow and don't uncover issues very frequently).
    
Add `make short` target which does build+testshort+checkshort.


I will send out a PSA when this goes in.
CC @tamird @jordanlewis @petermattis

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/13136)
<!-- Reviewable:end -->

RaduBerinde committed:
all: add testshort target that skips slow tests

Adding `make testshort` which adds the `-short` flag to `make test`. Disabling
tests over 5 seconds in this mode (except the main SQL logic test suite).

A reasonable workflow would be to run `make test PKG=./pkg/foo` for whatever
package you're touching and run `make testshort` for the rest.

Fixes #13025.

tamird commented on Makefile:
```diff
@@ -113,6 +113,9 @@ else
 	$(GO) test $(GOFLAGS) -tags '$(TAGS)' -run "$(TESTS)" -bench "$(BENCHES)" -timeout $(TESTTIMEOUT) $(PKG) $(TESTFLAGS)
 endif
 
+testshort: GOFLAGS += -short
```
testflags?

tamird replied:
Oh maybe that's not a thing.

RaduBerinde replied:
I think I can add it to TESTFLAGS just as well. I copied from testrace but in that case the race flag actually influences the build so it's different.

One thing though is if I do `TESTFLAGS += -short` and I also pass some flags like`make test TESTFLAGS='-v'`, the `-short` does not show up. I don't understand why.. This is also a problem with `make testslow`: if you run make testslow TESTFLAGS='-meh' the `-v` doesn't get passed.

tamird reviewed (approved):
LGTM

RaduBerinde commented:
Ah, I needed `override`. Updated.

tamird commented:
heh, TIL. We probably have other variables being mishandled in this way. LGTM.

RaduBerinde committed:
makefile: add override directives

Apparently '+=' doesn't do anything if the variable was set form the command
line. Prepending `override` does what we expect.

RaduBerinde commented:
I added another commit that does override for all our `+=`s. I tested a few of them (TAGS with make check, GOFLAGS with testrace).

RaduBerinde committed:
makefile: add checkshort and short targets

Add `make checkshort` which is `make check` with TestMetacheck and TestUnconvert
disabled (these tests are slow and don't uncover issues very frequently).

Add `make short` target which does build+testshort+checkshort.

RaduBerinde commented:
While I'm at it, I applied a similar treatment to `make check` and added a `make short` target that does `build+testshort+checkshort`.

tamird commented:
Neat.

On Jan 25, 2017 10:07, "RaduBerinde" <notifications@github.com> wrote:

> While I'm at it, I applied a similar treatment to make check and added a make
> short target that does build+testshort+checkshort.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/13136#issuecomment-275132218>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABdsPNp83XkzDZ3KHYHY0ydgsOKfZVq3ks5rV2VAgaJpZM4LtjNa>
> .
>
